packages/lit-node-client-nodejs/src/lib/lit-node-client-nodejs.ts

```diff
@@ -495,18 +497,66 @@ export class LitNodeClientNodeJs
    * will call refresh if the block hash is expired
    * @returns {Promise<string>} latest block hash from `handhsake` with the lit network.
    */
-  getLatestBlockhash = (): string => {
-    if (!this.ready) {
-      logError('Client not connected, remember to call connect');
+  getLatestBlockhash = async (): Promise<string> => {
+    const requestId = this.getRequestId();
+
+    this.connectedNodes = new Set();
+
+    if (this.config.bootstrapUrls.length <= 0) {
       throwError({
-        message: 'Client not connected',
-        errorKind: LIT_ERROR.LIT_NODE_CLIENT_NOT_READY_ERROR.kind,
-        errorCode: LIT_ERROR.LIT_NODE_CLIENT_NOT_READY_ERROR.code,
+        message: `Failed to get bootstrapUrls for network ${this.config.litNetwork}`,
+        errorKind: LIT_ERROR.INIT_ERROR.kind,
+        errorCode: LIT_ERROR.INIT_ERROR.name,
+      });
+    }
+
+    const respList = [];
+
+    for (const url of this.config.bootstrapUrls) {
+      const challenge = this.getRandomHexString(64);
+
+      try {
+        const resp = await this.handshakeWithNode(
+          { url, challenge },
+          requestId
+        );
+        respList.push(resp);
+      } catch (e) {
+        logError(`Error getting latestBlockhash from node ${url}`, e);
+      }
+    }
+
+    // -- if the list is empty, then return an error
+    if (respList.length === 0) {
+      throwError({
+        message: 'Failed to get latestBlockhash from any node',
+        errorKind: LIT_ERROR.INIT_ERROR.kind,
+        errorCode: LIT_ERROR.INIT_ERROR.name,
```

Comment:
Is the initialization check happening at other places? I assume the logic will remain the same? Let's extract it out in a function in that case?

Reply:
i think this getLatestBlockhash is that function?  Or are you thinking this has overlap with the connect() function which actually does the same thing (handshake w/all nodes) ?